lib/parser.js

```diff
@@ -49,6 +51,14 @@ exports.parse = function parse(source, options) {
     // Attach comments to the copy rather than the original.
     attachComments(comments, copy.program, lines);
 
+    // Attach inline sourcemap metadata when possible
+    Object.defineProperty(copy, 'inputSourceMap', {
```

Comment:
we could just a different way to annotate the ast, for now, I'm using this non-enumerable property.